joulescope_ui/range_tool.py

```diff
@@ -117,6 +117,7 @@ def iterate(self, samples_per_iteration=None):
             progress = self._worker.sigProgress.emit
         else:
             progress = self.progress
+        # Axel Jacobsen, AUG 1 added cast to int for samples_per_iteration to avoid floating point error on range
```

Comment:
There must be a better place to solve the issue (i.e. `samples_per_iteration` is a float when it should be an int)

Reply:
Agreed.  The RangeToolIterable.__init__ should enforce the cast, and we should compute it as an int to be clear.  Feel free to fix this, or I can take a look.

Reply:
Sounds good! I will fix this and push the update soon.